### DISCUSSION TITLE
Demonstration for Openbox-like behavior of `drawContents=false`

### DISCUSSION THREAD

**Discussion by tokyo4j** [labwc/labwc#1967] [Show and tell]
I implemented Openbox-like behavior of `drawContents=false`, where surface is clipped or white rectangle is appended to show the window geometry while resizing: tokyo4j/labwc@8116255

https://github.com/labwc/labwc/assets/22029524/2472ac77-9a2c-42f8-a388-e9b8e10aa230

I'm showing it here in discussion rather than creating a PR because its implementation is a bit too hacky and I don't think it's appropriate to merge it into master at the moment.

The parts in my implementation that I don't like are:
- It updates the position of `view->scene_tree` while resizing.
- It temporarily updates `view->current` then call `ssd_update_geometry()` to draw ssd that's out of sync with actual view geometry.

So, my implementation breaks assumption that the position of `view->scene_tree` is synced with `view->current` and `view->current` is the view geometry that's actually rendered. I'm not sure how much it affects the future development, but I don't think this feature is worth bringing such a semantical confusion into our codebase.

**Comment by Consolatis** [comment]
Couldn't we just do something like
```c
wlr_scene_node_set_enabled(view->scene_node, false);
wlr_scene_node_set_enabled(view->background, true);
```

when starting the interactive resize, then resize `view->background` during the resize and at the end toggle between background and surface contents again? That would hide the existing surface contents but that feels cleaner to me anyway.

Also I didn't know that subsurface_tree_clip() actually works for random scene trees and not just subsurface trees. Nice!

> So, my implementation breaks assumption that the position of `view->scene_tree` is synced with `view->current` and `view->current` is the view geometry that's actually rendered. I'm not sure how much it affects the future development, but I don't think this feature is worth bringing such a semantical confusion into our codebase.

Agree.

**Reply by tokyo4j** [reply]
Your idea may be cleaner, but I don't think hiding contents is a good idea. I'd rather prefer the current behavior that just shows outlines, because sometimes the window geometry after resize is based on the original geometry or its contents.

I'm now thinking of adopting the same approach as snapping overlay which shows outlines or translucent background. I slightly prefer Openbox's behavior, but I think it'll make the geometry more visible than only outlines. Also, AFAIR it's the way xfce4 took.

EDIT:
The reason why I didn't implement it in the earlier PR is that we need to care about flickering, disabling translucent background for pixman renderer, and added theme settings (`bg.enable`/`border.enable`/`bg.color`/`border.color`/`border.width`) like I did for snapping overlay.

recording from xfce4:

https://github.com/user-attachments/assets/3018c553-6450-40d7-9c5f-783e71522622

**Reply by johanmalm** [reply]
Agree. Sounds good
